packages/malloy/src/lang/ast/parameters/has-parameter.ts

```diff
@@ -84,6 +90,22 @@ export class HasParameter extends MalloyElement {
           };
         }
       }
+      if (constant.type === 'filter expression') {
+        if (this.type !== 'filter expression') {
+          this.logError(
+            'parameter-missing-default-or-type',
+            `Filter expression parameters must have expicit filter type, for example '${this.name}::filter<string>'`
+          );
+        }
+        if (this.filterType) {
+          checkFilterExpression(this, this.filterType, constant.value);
+        }
+        return {
+          value: constant.value,
+          name: this.name,
+          type: constant.type,
+        };
+      }
```

Comment:
I would sort of expect, like how you can say `source: x(foo is null::string)` to also be able to say `source: x(foo is f'foo'::filter<string>)` (as well as `source: x(foo::filter<string> is f'foo')`). It doesn't seem like this allows for that?